### DISCUSSION TITLE
🎉 Announcing Interactivity in @lottiefiles/dotlottie-web v0.30.2 🎉

### DISCUSSION THREAD

**Discussion by samuelOsborne** [LottieFiles/dotlottie-web#313] [Announcements]
We're excited to bring to you dotLottie **Interactivity**! With this release you can now include interactivity definition files inside your .lottie files and load them at runtime across our new Web, iOS and Android players. Powered by state machines, it's possible to create deeply interactive scenarios for your animations. To start creating, use our [dotlottie-js](https://developers.lottiefiles.com/docs/tools/dotlottie-js/interactivity/interactive-scenarios/) library.

#### New Features

  1. **State Machine Integration**🎸
     - DotLottie **`loadStateMachine` Method**: Load a state machine directly from a .lottie file.
     - DotLottie **`startStateMachine` Method**: Start the loaded state machine.
     - DotLottie **`stopStateMachine`. Method**: Cleanly end the started state machine.
     - DotLottie **`postStateMachineEvent` Method**: Send events to the state machine to drive animation behavior.

  ##### Available Events

  ```js
  const dotLottie = new DotLottie(...);

  const smLoaded = dotLottie.loadStateMachine("some_state_machine_id"); // Load a state machine from the .lottie file
  const smStarted = dotLottie.startStateMachine();                // Start the loaded state machine

  dotLottie.postStateMachineEvent("Bool: true");         // Post a boolean event
  dotLottie.postStateMachineEvent("Bool: false");        // Post a boolean event
  dotLottie.postStateMachineEvent("String: ...");        // Post a string event
  dotLottie.postStateMachineEvent("Numeric: 0.0");       // Post a numeric event
  dotLottie.postStateMachineEvent("OnPointerDown: 0.0 0.0"); // Post a pointer down event
  dotLottie.postStateMachineEvent("OnPointerUp: 0.0 0.0");   // Post a pointer up event
  dotLottie.postStateMachineEvent("OnPointerMove: 0.0 0.0"); // Post a pointer move event
  dotLottie.postStateMachineEvent("OnPointerEnter: 0.0 0.0"); // Post a pointer enter event
  dotLottie.postStateMachineEvent("OnPointerExit: 0.0 0.0");  // Post a pointer exit event
  dotLottie.postStateMachineEvent("OnComplete");             // Post a complete event
  dotLottie.postStateMachineEvent("SetNumericContext: [key] [value]");  // Update the numeric context

   const smStopped = dotLottie.stopStateMachine(); // End the started state machine
  ```

  > `DotLottie` will automatically set up and clean up the pointer events on the provided canvas and other animation
  > playback events for you when you call `startStateMachine` and `stopStateMachine`.
  More details can be found in the
  [LottieFiles developers portal](https://developers.lottiefiles.com/docs/dotlottie-player/)

**Comment by samuelOsborne** [comment]
Heres some examples of it in use ✨ 

https://codepen.io/lottiefiles/pen/VwJvdRN

**Reply by jprabawa** [reply]
Is the example still working? The animations dont seem to be playing

**Reply by theashraf** [reply]
@jprabawa This feature is still experimental. I’ve written an updated guide as a reference:
[https://github.com/LottieFiles/dotlottie-web/wiki/dotLottie-State-Machine-Guide](https://github.com/LottieFiles/dotlottie-web/wiki/dotLottie-State-Machine-Guide)

I’m happy to answer any questions, and it would be great to get your feedback and hear about your use cases.

**Reply by jprabawa** [reply]
Thanks @theashraf, it would be great to have a codepen example to accompany the guide to have a better understanding of how things actually work in action. Was quite hopeful with the codepen from @samuelOsborne but i dont think its working correctly?

**Reply by theashraf** [reply]
@jprabawa wiki updated with running codepens, please let me know if you're facing any issues

**Reply by jprabawa** [reply]
Thank you!
